app/services/DraftMovementService.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import connectors.emcsTfe.DraftMovementConnector
+import models.response.emcsTfe.messages.submissionFailure.GetSubmissionFailureMessageResponse
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.Logging
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class DraftMovementService @Inject()(draftMovementConnector: DraftMovementConnector) extends Logging {
+
+  def putErrorMessagesAndMarkMovementAsDraft(ern: String, getSubmissionFailureMessageResponse: GetSubmissionFailureMessageResponse)
+                                            (implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Option[String]] = {
+    val correlationId = getSubmissionFailureMessageResponse.ie704.header.correlationIdentifier.getOrElse("")
```

Comment:
Would it be better to `flatMap` on this rather than having a `getOrElse("")` which seems a bit out of place. As the signature is already a `Future[Option]` So, I think you'd be able to `Future.sequence` this so that the `option` appears within the future and then you can just flatten it down - if that makes sense?